### PR TITLE
Test consistency

### PR DESCRIPTION
**Pull request by DaddyWesker**
Ok, so I've added test consistency. Currently it is failing on the last line of the test_consistency method since `getNearEntities` returns Oak Log instead of oak_log (which is looks like this everywhere else). So, it needs some changes i guess.  I'll think about it.

Also there are some changes to visualization - now it works in the main thread. 

This PR won't works only with my [Vereya PR ](https://github.com/trueagi-io/Vereya/pull/45)

### CODE REVIEW

**Review by CICS-Oleg — COMMENTED** [review]
Why do you think it's unnecessary?

File: mcdemoaux/vision/vis.py [hunk 1]
```diff
@@ -22,10 +22,6 @@ def run(self):
             while self.queue:
                 with self._lock:
                     data = self.queue.pop()
-                    if len(data[1].shape) == 3:
-                        image = cv2.cvtColor(data[1], cv2.COLOR_BGR2RGB)
```
Comment (CICS-Oleg) [inline mcdemoaux/vision/vis.py:26]: Why do you think it's unnecessary?

Comment (CICS-Oleg) [reply mcdemoaux/vision/vis.py:26]: Or do we have now new format and new conversion with new image transfer protocol?

Comment (DaddyWesker) [reply mcdemoaux/vision/vis.py:26]: @CICS-Oleg  well, I'm not sure actually. Does we need that if-statement? Currently bgr2rgb done in timestamped_video_frame in my PR https://github.com/trueagi-io/minecraft-demo/pull/136/files which still waiting for your approval =)

**Comment by DaddyWesker** [comment]
So, I've deleted here all my changes related to visualizer. So now this PR is only about test_consistency.

File: tagilmo/utils/vereya_wrapper.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ def observeProc(self, nAgent=None):
             self.worldStates[n] = self.agent_hosts[n].getWorldState()
             self.isAlive[n] = self.worldStates[n].is_mission_running
             obs = self.worldStates[n].observations
-            self.observe[n] = json.loads(obs[-1].text.replace("minecraft:", "")) if len(obs) > 0 else None
+            self.observe[n] = json.loads(obs[-1].text.replace("minecraft:", "").replace("block.minecraft.", "").replace("item.minecraft.", "")) if len(obs) > 0 else None
```
[the remark this answers is not in the document]
Comment (DaddyWesker) [reply tagilmo/utils/vereya_wrapper.py]: @Necr0x0Der  I've changed that one. Though it now done through for loop and I'm not sure it is better solution than three replace for one string.

File: tagilmo/utils/vereya_wrapper.py [hunk 3]
```diff
@@ -237,7 +237,12 @@ def isLineOfSightAvailable(self, nAgent=0):
         return not self.observe[nAgent] is None and 'LineOfSight' in self.observe[nAgent]
 
     def getLineOfSights(self, nAgent=0):
-        return self.observe[nAgent]['LineOfSight'] if self.isLineOfSightAvailable(nAgent) else None
+        if self.isLineOfSightAvailable(nAgent):
+            los = self.observe[nAgent]['LineOfSight']
+            los['type'] = los['type'].replace("minecraft:", "")
+            return los
+        return None
+        # return self.observe[nAgent]['LineOfSight'] if self.isLineOfSightAvailable(nAgent) else None
```
Comment (noskill) [inline tagilmo/utils/vereya_wrapper.py:245]:
```suggestion
```

File: tests/vereya/test_consistency.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+import logging
+import json
+import time
+from tagilmo import VereyaPython
+import tagilmo.utils.mission_builder as mb
+from tagilmo.utils.vereya_wrapper import MCConnector, RobustObserver
+from base_test import BaseTest
+
+
+def init_mission(mc, start_x=None, start_y=None, start_z=None):
```
Comment (noskill) [inline tests/vereya/test_consistency.py:11]: if this function has nothing specific you can replace it with init_mission from common.py

Comment (DaddyWesker) [reply tests/vereya/test_consistency.py:11]: @noskill  it does differ

File: tagilmo/utils/vereya_wrapper.py [hunk 5]
```diff
@@ -436,11 +440,22 @@ def observeProcCached(self):
...
 
+    def cleanRecipes(self, recipes):
+        if recipes is not None:
+            for i in range(len(recipes)):
+                recipes[i]['name'] = recipes[i]['name'].replace("block.minecraft.", "").replace("item.minecraft.", "")
+                recipes[i]['ingredients'] = list(filter(None, recipes[i]['ingredients']))
+                for j in range(len(recipes[i]['ingredients'])):
+                    if len(recipes[i]['ingredients'][j][0]) <= 0:
+                        continue
+                    recipes[i]['ingredients'][j][0]['type'] = recipes[i]['ingredients'][j][0]['type'].replace("block.minecraft.", "").replace("item.minecraft.", "")
```
Comment (Necr0x0Der) [inline tagilmo/utils/vereya_wrapper.py:451]: Why only one alternative of ingredients is considered?
Also, I'm not sure if it is really better to explicitly take the structure of receipts into account, or to use a generic function like this:
```
    def remove_mcprefix_rec(data):
        if isinstance(data, str):
            return data.split('.')[-1] if 'minecraft' in data else data
        if isinstance(data, dict):
            r = {}
            for k in data.keys():
                r[remove_mcprefix_rec(k)] = remove_mcprefix_rec(data[k])
            return r
        if isinstance(data, list):
            return remove_mcprefix_rec(l) for l in data]
        return data
```

Comment (DaddyWesker) [reply tagilmo/utils/vereya_wrapper.py]: @Necr0x0Der works like a charm. Thanks. I've added to PR

**Review by Necr0x0Der — APPROVED** [review]

**Pull request merged**